js/core/structure.js

```diff
@@ -97,13 +97,19 @@ define(
                 if (!conf.noTOC) {
                     var $ul = makeTOCAtLevel($("body", doc), doc, [0], 1, conf);
                     if (!$ul) return;
-                    var $sec = $("<section id='toc'/>").append("<h2 class='introductory'>" + conf.l10n.toc + "</h2>")
+                    var w = ( conf.useExperimentalStyles ? "nav" : "section" );
+                    var $sec = $("<" + w + " id='toc'/>").append("<h2 class='introductory'>" + conf.l10n.toc + "</h2>")
```

Comment:
Nit (though I would prefer we only indent with 2 spaces... will eventually fix ReSpec's horrible whitespace conventions):
```JS
                    var $sec = $("<" + w + " id='toc'/>")
                                                        .append("<h2 class='introductory'>" +  conf.l10n.toc + "</h2>")
                                                        .append($ul)
```